src/Microsoft.AspNet.Hosting/Server/ServerLoader.cs

```diff
@@ -18,59 +18,29 @@ public ServerLoader(IServiceProvider services)
             _services = services;
         }
 
-        public IServerFactory LoadServerFactory(string serverFactoryIdentifier)
+        public IServerFactory LoadServerFactory([NotNull] string assemblyName)
         {
-            if (string.IsNullOrEmpty(serverFactoryIdentifier))
+            if (string.IsNullOrEmpty(assemblyName))
             {
-                throw new ArgumentException(string.Empty, nameof(serverFactoryIdentifier));
+                throw new ArgumentException(string.Empty, nameof(assemblyName));
```

Comment:
Both, NotNull doesn't yet enforce anything at runtime, and no it won't prevent empty strings